### PR TITLE
Share log as file instead of just text

### LINKED ISSUE #2205: log should better be shared as a file
with https://github.com/deltachat/deltachat-ios/pull/2126 we have the option to share a log, which is great!

however, the log is shared as text, which is unwieldy and error prone. we went through that on android recently :)

better share the log as a file, this allows the user also to take some notes - wrt editing: this is possible sufficiently beforehand

### PR DESCRIPTION
**Pull request by zeitschlag**
When sharing, log-text and notes get written into a file `log.txt` which gets shared then.

Closes #2205.

### CODE REVIEW

File: deltachat-ios/Helper/Utils.swift [hunk 1]
```diff
@@ -82,16 +82,25 @@ struct Utils {
...
+
+        Utils.share(url: url, parentViewController: parentViewController, sourceItem: sourceItem)
     }
 
     public static func share(text: String, parentViewController: UIViewController, sourceItem: UIBarButtonItem) {
-        let activityVC = UIActivityViewController(activityItems: [text], applicationActivities: nil)
-        activityVC.popoverPresentationController?.barButtonItem = sourceItem // iPad crashes without a source
+        guard let data = text.data(using: .utf8) else { return }
+
+        let tempLogfileURL = FileManager.default.localDocumentsDir.appendingPathComponent("log.txt")
```
Comment (r10s) [inline deltachat-ios/Helper/Utils.swift:93]: why not use `temporaryDirectory`, that seems to be a better fit that saving the file "forever" in the app library directory beside the database. 

```suggestion
        let tempLogfileURL = FileManager.default. temporaryDirectory.appendingPathComponent("deltachat-log.txt")
```

i also made the file name a bit more verbose (this is also the prefix we use on android)

**Review by r10s — COMMENTED** [review]
sth. is wrong with the encoding, when trying to save log.txt to the "Notes App", all unicode characters are broken:



which is weird, as `.utf8` is stated explicitly

File: deltachat-ios/Helper/Utils.swift (hunk 1 above)
Comment (zeitschlag) [reply deltachat-ios/Helper/Utils.swift:93]: Good points, thank you! 

Using the temporary directory for that just didn't come to my mind, but you're right: It's a better fit.

**Comment by zeitschlag** [comment]
This is interesting, as when I open `deltachat-log.txt` on a Mac with Preview.app or BBEdit or TextEdit.app, I see the emojis:

Screenshot 2024-08-21 at 10 07 34

So at this point I'd blame Notes or Safari on iOS?

File: deltachat-ios/Helper/Utils.swift [hunk 2]
```diff
@@ -82,16 +82,25 @@ struct Utils {
...
-        }
+        guard let url = URL(string: url) else { return }
+
+        Utils.share(url: url, parentViewController: parentViewController, sourceItem: sourceItem)
     }
 
     public static func share(text: String, parentViewController: UIViewController, sourceItem: UIBarButtonItem) {
-        let activityVC = UIActivityViewController(activityItems: [text], applicationActivities: nil)
-        activityVC.popoverPresentationController?.barButtonItem = sourceItem // iPad crashes without a source
+        guard let data = text.data(using: .unicode) else { return }
```
Comment (r10s) [inline deltachat-ios/Helper/Utils.swift:91]: `.unicode` works in general, however, generates UTF-16/USC-2 instead of UTF-8. this can result in issues on the receiving side, as UTF-16/USC-2 is not as widely supported as UTF-8.

however, this is no longer a blocker, i'll have a rought look if there isn't just a BOM missing or so.

File: DcCore/DcCore/Helper/DatabaseHelper.swift [hunk 3]
```diff
@@ -14,12 +16,7 @@ public class DatabaseHelper {
     }
 
     var localDbFile: String {
-        return localDocumentsDir.appendingPathComponent("messenger.db").path
-    }
-
-    var localDocumentsDir: URL {
-        let paths = NSSearchPathForDirectoriesInDomains(.libraryDirectory, .userDomainMask, true)
-        return URL(fileURLWithPath: paths[0], isDirectory: true)
+        return FileManager.default.localDocumentsDir.appendingPathComponent("messenger.db").path
```
Comment (r10s) [inline DcCore/DcCore/Helper/DatabaseHelper.swift:19]: this unrelated change,  touching a _very_ sensible area,  is no longer needed. i think, the additional abstraction is not really helpful without code being reused.

but no blocker.

**Review by r10s — APPROVED** [review]
please double check if we still want to have the unrelated changes in opening main database in.

otherwise lgtm

File: deltachat-ios/Helper/Utils.swift (hunk 2 above)
Comment (r10s) [reply deltachat-ios/Helper/Utils.swift:91]: indeed, when adding a BOM, UTF-8 works as expected.

background: BOM (Byte-Order-Mark) is not _needed_ for an UTF-8 file, however, many apps and libs need that at the beginning of a UTF-8 file to identify it as such (otherwise, a UTF-8 file content looks same to an ASCII/ANSI file ... and, without having metadata, can only be distinguished by some heuristics - which some libs do and others do not)

```suggestion
        guard var data = text.data(using: .utf8) else { return }
        let bom = Data([0xEF, 0xBB, 0xBF])
        data.insert(contentsOf: bom, at: 0)
```

with this change, UTF-8 works in "Notes" pad as well as in "WhatsApp"
